examples/nodegroup-py/__main__.py

```diff
@@ -28,7 +28,10 @@
     instance_profile=instance_profile0,
 )
 
-clusterK8sProvider = k8s.Provider("clusterK8sProvider", kubeconfig=cluster.kubeconfig.apply(lambda k: json.dumps(k)))
+clusterK8sProvider = k8s.Provider(
+    "clusterK8sProvider",
+    kubeconfig=cluster.kubeconfig.apply(lambda k: json.dumps(k, sort_keys=True)),
```

Comment:
Why does `sort_keys` matter? Does it change every time otherwise? Or has this just shown up because we're moving things around?

Reply:
It was changing, causing the provider to be recreated along with all the resources using that provider. It wouldn't always happen, but fairly consistently when running `pulumi up` multiple times.

An improvement over having users do this would be #462, adding a `kubeconfigJSON` output, which would return the canonicalized (sorted) JSON string, so users wouldn't have to deal with converting the object to JSON and keeping it sorted. I can add a commit to this PR that adds that.